packages/model-viewer/src/features/scene-graph/material.ts

```diff
@@ -13,20 +13,116 @@
  * limitations under the License.
  */
 
-import {MeshStandardMaterial, Texture as ThreeTexture} from 'three';
+import {MeshStandardMaterial, PixelFormat, Texture as ThreeTexture, TextureEncoding} from 'three';
 
-import {GLTF, Material as GLTFMaterial} from '../../three-components/gltf-instance/gltf-2.0.js';
+import {Asset, GLTF, Image as GLTFImage, Material as GLTFMaterial, Sampler as GLTFSampler, Texture as GLTFTexture, TextureInfo as GLTFTextureInfo} from '../../three-components/gltf-instance/gltf-2.0.js';
 
 import {Material as MaterialInterface, RGB} from './api.js';
 import {PBRMetallicRoughness} from './pbr-metallic-roughness.js';
 import {TextureInfo} from './texture-info.js';
 import {$correlatedObjects, $onUpdate, $sourceObject, ThreeDOMElement} from './three-dom-element.js';
 
 
+
 const $pbrMetallicRoughness = Symbol('pbrMetallicRoughness');
 const $normalTexture = Symbol('normalTexture');
 const $occlusionTexture = Symbol('occlusionTexture');
 const $emissiveTexture = Symbol('emissiveTexture');
+const $backingThreeMaterial = Symbol('backingThreeMaterial');
+export const $provideApplicator = Symbol('TextureApplicator');
+export const $material = Symbol('material');
+export const $threeTexture = Symbol('threeTexture');
+export const $usage = Symbol('usage');
+export const $encoding = Symbol('encoding');
+export const $format = Symbol('format');
+export const $gltf = Symbol('gltf');
+export const $gltfTextureInfo = Symbol('gltfTextureInfo');
+export const $gltfTexture = Symbol('gltfTexture');
+export const $gltfSampler = Symbol('gltfSampler');
+export const $gltfImage = Symbol('gltfImage');
+
+// Defines what a texture will be used for.
+export enum TextureUsage {
+  Base,
+  Metallic,
+  Normal,
+  Occlusion,
+  Emissive,
+}
+
+/**
+ * TextureContext holds the glTF and Three data needed for creating and updating
+ * a texture in one place.
+ */
+export class TextureContext {
+  [$gltf]: GLTF;
+  [$material]: MeshStandardMaterial|null;
+  [$threeTexture]: ThreeTexture|null = null;
+  [$gltfTextureInfo]: GLTFTextureInfo;
+  [$gltfTexture]: GLTFTexture;
+  [$gltfSampler]: GLTFSampler;
+  [$gltfImage]: GLTFImage;
+  [$usage]: TextureUsage;
+  [$encoding]: TextureEncoding;
+  [$format]: PixelFormat;
```

Comment:
Is this used? I believe we aren't allowed to change format from what the file specifies.

Reply:
oh yeah, good catch